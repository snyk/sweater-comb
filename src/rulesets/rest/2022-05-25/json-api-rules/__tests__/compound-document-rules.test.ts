import { OpenAPIV3 } from "@useoptic/openapi-utilities";
import { RuleRunner, TestHelpers } from "@useoptic/rulesets-base";
import { context } from "../../__tests__/fixtures";
import { compoundDocuments } from "../compound-document-rules";

const baseJson = TestHelpers.createEmptySpec();

describe("compoundDocuments", () => {
  test("fails when specifying compound documents", () => {
    const afterJson: OpenAPIV3.Document = {
      ...baseJson,
      paths: {
        "/api/example": {
          get: {
            responses: {
              "200": {
                description: "",
                content: {
                  "application/vnd.api+json": {
                    schema: {
                      properties: {
                        included: {
                          type: "object",
                          properties: {},
                        },
                      },
                    },
                  },
                },
              },
            },
          },
        },
      },
    };

    const ruleRunner = new RuleRunner([compoundDocuments]);
    const ruleInputs = {
      ...TestHelpers.createRuleInputs(baseJson, afterJson),
      context,
    };
    const results = ruleRunner.runRulesWithFacts(ruleInputs);
    expect(results.length).toBeGreaterThan(0);
    expect(results.every((result) => result.passed)).toBe(false);
    expect(results).toMatchSnapshot();
  });
});
