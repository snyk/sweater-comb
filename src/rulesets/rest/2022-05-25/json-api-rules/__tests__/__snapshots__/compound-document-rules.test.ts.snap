// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compoundDocuments fails when specifying compound documents 1`] = `
[
  {
    "change": {
      "added": {
        "contentType": "application/vnd.api+json",
        "flatSchema": {},
      },
      "changeType": "added",
      "location": {
        "conceptualLocation": {
          "inResponse": {
            "body": {
              "contentType": "application/vnd.api+json",
            },
            "statusCode": "200",
          },
          "method": "get",
          "path": "/api/example",
        },
        "conceptualPath": [
          "operations",
          "/api/example",
          "get",
          "responses",
          "200",
          "application/vnd.api+json",
        ],
        "jsonPath": "/paths/~1api~1example/get/responses/200/content/application~1vnd.api+json",
        "kind": "body",
      },
    },
    "condition": undefined,
    "docsLink": undefined,
    "error": "Expected to not find a partial match",
    "exempted": false,
    "expected": "{"schema":{"properties":{"included":{}}}}",
    "isMust": true,
    "isShould": false,
    "name": "disallow compound documents",
    "passed": false,
    "received": "{"schema":{"properties":{"included":{"type":"object","properties":{}}}}}",
    "type": "added",
    "where": "GET /api/example response 200 response body: application/vnd.api+json",
  },
]
`;
